ts/src/kuna.ts

```diff
@@ -1000,19 +1923,38 @@ export default class kuna extends Exchange {
                 }
             }
         }
+        if (body !== undefined) {
+            body = JSON.stringify (body);
+        }
         return { 'url': url, 'method': method, 'body': body, 'headers': headers };
     }
 
     handleErrors (code, reason, url, method, headers, body, response, requestHeaders, requestBody) {
-        if (response === undefined) {
+        //
+        //    {
+        //        "errors": [
+        //            {
+        //                "extensions": {
+        //                    "code": "IP_NOT_IN_WHITE_LIST"
+        //                },
+        //                "code": "IP_NOT_IN_WHITE_LIST"
+        //            }
+        //        ]
+        //    }
+        //
+        const errors = this.safeValue (response, 'errors');
+        if ((response === undefined) && (errors === undefined)) {
             return undefined;
         }
-        if (code === 400) {
-            const error = this.safeValue (response, 'error');
+        if ((errors !== undefined) || (code === 400)) {
+            let error = this.safeValue (errors, 0);
+            if (error === undefined) {
+                error = this.safeValue (response, 'error');
+            }
             const errorCode = this.safeString (error, 'code');
             const feedback = this.id + ' ' + this.json (response);
             this.throwExactlyMatchedException (this.exceptions, errorCode, feedback);
-            // fallback to default error handler
+            throw new ExchangeError (feedback);
```

Comment:
the exceptions section seems very incomplete, maybe we could try to improve it?
![image](https://github.com/ccxt/ccxt/assets/43336371/f2949845-72bc-4d62-97f8-ef62bc4129c4)